drivers/include/lis2dh12.h

```diff
@@ -231,6 +257,47 @@ int lis2dh12_set_int(const lis2dh12_t *dev, const lis2dh12_int_params_t *params,
  * @return  LIS2DH12_NOBUS on bus errors
  */
 int lis2dh12_read_int_src(const lis2dh12_t *dev, uint8_t *data, uint8_t int_line);
+
+/**
+ * @brief   Set the FIFO configuration
+ *
+ * @param[in] dev       device descriptor
+ * @param[in] config    device FIFO configuration
+ *
+ * @return  LIS2DH12_OK on success
+ */
+int lis2dh12_set_fifo(const lis2dh12_t *dev, lis2dh12_fifo_t *config);
+
+/**
+ * @brief   Restart the FIFO mode
+ *          this sets the FIFO mode in BYPASS mode and then back to previous mode
+ *
+ * @param[in] dev       device descriptor
+ *
+ * @return  LIS2DH12_OK on success
+ */
+int lis2dh12_restart_fifo(const lis2dh12_t *dev);
+
+/**
+ * @brief   Read the FIFO source register
+ *
+ * @param[in] dev       device descriptor
+ * @param[out] data     register values
+ *
+ * @return  LIS2DH12_OK on success
+ */
+int lis2dh12_read_fifo_src(const lis2dh12_t *dev, uint8_t *data);
+
+/**
+ * @brief   Read the FIFO data
+ *
+ * @param[in] dev       device descriptor
+ * @param[out] data     FIFO data
+ * @param[in] number    amount of FIFO data to be read
```

Comment:
What happens if there is less data in the fifo than `number`?

Reply:
The module will return 0 in this registers, so this shouldn't be an issue.

Reply:
But how can the user tell this apart from valid data?
Can we return the number of valid records on success?

Reply:
The `FIFO_SRC_REG` contains the number of unread fifo data. We can check this before and after.
But should this be part of the user, so he can check weather the number is available or not?

Reply:
The user shouldn't have to interpret register values, that's the task of the driver.
Instead of having two API functions

 - get number of fifo events
 - read n events from fifo

we could just have

 - read up to n events from fifo, return number of events read (`min(n, FIFO_SRC_REG)`)

Is there any case where you would want to know the number of events in the fifo without also reading out the events?

Reply:
ah okay, no I think there is no case.